### DISCUSSION TITLE
Why "[" getting added to the sql query ?

### DISCUSSION THREAD

**Discussion by assinchu** [vanna-ai/vanna#251] [Q&A]
SELECT JOB\_STATUS FROM ACT\_INFO WHERE ID = 956060;[

{"model":"mistral:7b","created_at":"2024-02-21T21:51:06.787112465Z","message":{"role":"assistant","content":"SELECT JOB\\_STATUS FROM ACT\\_INFO WHERE ID = 956060;[\n\nThis query should return the status of the job with an ID of 956060 from the ACT\\_INFO table."},"done":true,"total_duration":1972669325,"load_duration":394033,"prompt_eval_count":1895,"prompt_eval_duration":1108901000,"eval_count":51,"eval_duration":862615000}
SELECT JOB\_STATUS FROM ACT\_INFO WHERE ID = 956060;[

**Comment by zainhoda** [comment]
Mistral can be a little fickle with its output. Unfortunately that's just the random nature of LLMs.

Can you try overriding your `generate_sql` method like we have here and adding your own rules about what characters to remove 
https://github.com/vanna-ai/vanna/blob/main/src/vanna/ollama/__init__.py#L49-L58

**Reply by assinchu** [reply]
Thanks. Let me try this.
BTW, i tried llama13B, i dont see this issue.
